metrics/chrf/chrf.py

```diff
@@ -0,0 +1,152 @@
+    title = "chr{F}++: words helping character n-grams",
+    author = "Popovi{\'c}, Maja",
+    booktitle = "Proceedings of the Second Conference on Machine Translation",
+    month = sep,
+    year = "2017",
+    address = "Copenhagen, Denmark",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/W17-4770",
+    doi = "10.18653/v1/W17-4770",
+    pages = "612--618",
+}
+@inproceedings{post-2018-call,
+    title = "A Call for Clarity in Reporting {BLEU} Scores",
+    author = "Post, Matt",
+    booktitle = "Proceedings of the Third Conference on Machine Translation: Research Papers",
+    month = oct,
+    year = "2018",
+    address = "Belgium, Brussels",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/W18-6319",
+    pages = "186--191",
+}
+"""
+
+_DESCRIPTION = """\
+ChrF and ChrF++ are two MT evaluation metrics. They both use the F-score statistic for character n-gram matches,
+and ChrF++ adds word n-grams as well which correlates more strongly with direct assessment. We use the implementation
+that is already present in sacrebleu.
+
+The implementation here is slightly different from sacrebleu in terms of the required input format. The length of
+the references and hypotheses lists need to be the same, so you may need to transpose your references compared to
+sacrebleu's required input format. See https://github.com/huggingface/datasets/issues/3154#issuecomment-950746534
+
+See the README.md file at https://github.com/mjpost/sacreBLEU#chrf--chrf for more information.
+"""
+
+_KWARGS_DESCRIPTION = """
+Produces ChrF(++) scores for hypotheses given reference translations.
+
+Args:
+    predictions: The system stream (a sequence of segments).
+    references: A list of one or more reference streams (each a sequence of segments).
+    char_order: Character n-gram order.
+    word_order: Word n-gram order. If equals to 2, the metric is referred to as chrF++.
+    beta: Determine the importance of recall w.r.t precision.
+    lowercase: Enable case-insensitivity.
+    whitespace: If `True`, include whitespaces when extracting character n-grams.
+    eps_smoothing: If `True`, applies epsilon smoothing similar
+    to reference chrF++.py, NLTK and Moses implementations. Otherwise,
+    it takes into account effective match order similar to sacreBLEU < 2.0.0.
+
+Returns:
+    'score': The chrF (chrF++) score,
+    'char_order': The character n-gram order,
+    'word_order': The word n-gram order. If equals to 2, the metric is referred to as chrF++,
+    'beta': Determine the importance of recall w.r.t precision
+
+Examples:
+
+    >>> prediction = ["The relationship between Obama and Netanyahu is not exactly friendly."]
+    >>> reference = [["The ties between Obama and Netanyahu are not particularly friendly."]]
+    >>> chrf = datasets.load_metric("chrf")
+    >>> results = chrf.compute(predictions=prediction, references=reference)
```

Comment:
As in TER I think we can add one line for doctest:
```suggestion
    >>> results = chrf.compute(predictions=prediction, references=reference)
    >>> print(results)
    {'score': 61.576379378113785, 'char_order': 6, 'word_order': 0, 'beta': 2}
```